src/stores/index.js

```diff
@@ -882,7 +942,14 @@ const Store = Reflux.createStore({
         null, // No error connecting.
         connectedDataService
       );
+
+      // Compass relies on `compass-connect` showing a progress
+      // bar, which is hidden after the instance information is loaded
+      // in another plugin.
+      this.StatusActions.showIndeterminateProgressBar();
     } catch (error) {
+      this.connectingConnectionAttempt = null;
+
       this.setState({
```

Comment:
don't we need a `finally` that also resets the state.isConnecting etc?

Reply:
Added a finally to the top level connect. 👍